### PR TITLE
[KOGITO-5120] Improve GitHub Action configuration

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/KOGITO-5120

### Details
- Simplified configuration
- Created `BUILD_MVN_OPTS` env variable to specify common params. This configuration will be updated as soon as a similar feature will be added to github action chain ( to have something similar to https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/40a61ce094719595ed103cfbb87ab44e3c40d61d/.ci/pull-request-config.yaml#L5 )
- Introduced `action-surefire-report` to parse test result and provide a test report

PR list
- https://github.com/kiegroup/kogito-pipelines/pull/160
- https://github.com/kiegroup/kogito-runtimes/pull/1292
- https://github.com/kiegroup/kogito-apps/pull/814
- https://github.com/kiegroup/kogito-examples/pull/681

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

File: .github/workflows/apps-pr.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.4
```
Comment (Ginxo) [inline .github/workflows/apps-pr.yml:28]:
```suggestion
        uses: kiegroup/github-action-build-chain@v2.5.4
```

File: .github/workflows/ci-pr.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.4
```
Comment (Ginxo) [inline .github/workflows/ci-pr.yml:28]:
```suggestion
        uses: kiegroup/github-action-build-chain@v2.5.4
```

File: .github/workflows/examples-pr.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.4
```
Comment (Ginxo) [inline .github/workflows/examples-pr.yml:28]:
```suggestion
        uses: kiegroup/github-action-build-chain@v2.5.4
```

File: .github/workflows/optaplanner-pr.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build Chain ${{ matrix.java-version }}
+        id: build-chain
+        uses: kiegroup/github-action-build-chain@v2.4
```
Comment (Ginxo) [inline .github/workflows/optaplanner-pr.yml:28]:
```suggestion
        uses: kiegroup/github-action-build-chain@v2.5.4
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check

**Comment by radtriste** [comment]
do we really want to run all 4 in parallel ? this is quite resources consuming and we are already limited on resources for GHAction ...

**Comment by radtriste** [comment]
do we really want to run all 4 in parallel ? this is quite resources consuming and we are already limited on resources for GHAction ...
But I do like the idea :)

**Comment by danielezonca** [comment]
> do we really want to run all 4 in parallel ? this is quite resources consuming and we are already limited on resources for GHAction ...
> But I do like the idea :)

It is true that it creates more builds but the aggregated execution time is similar: the overhead of each `compile install -DskipTests` for upstream repos of each job is like ~5 minutes.
In this way each build is way faster and even more important lightweight in terms of disk usage so I don't need anymore to cleanup the executor as first step (quite error prone script because it prevents us to simply take `ubuntu-latest`).
I would like to try this setup and see if it is stable, if so we can consider to partially merge them again (I think it could make sense only for OptaPlanner + Runtimes 🤔 )

**Pull request merged**